### ISSUE TITLE
Can't unenroll beats

### ISSUE THREAD

**Issue by clintongormley** [elastic/beats#9452]
In 6.5.1, I used the UI to "unenroll" my beat, which appeared to work from the UI, but the beat just reported:

> error retriving new configurations, will use cached ones: Beat "0550147b-ff0a-4c1b-beda-f8e2372cd797" not found

and continued with the last downloaded configuration.

**Comment by ph** [comment]
After talking with @mattapperson, the API does the right thing and return a 404, The current management workflow handles all the API errors the same way and recover from the last saved pulled config. In the case of a 404 we should instead treat it as we receive an empty configuration and show a message like this:

`if you re-enroll your beats you will have access to your configuration.`

I am not sure if we should stop the polling of the config when this happens it some circumstances it could be transient.

**Comment by mattapperson** [comment]
Maybe we should return JSON of “{ found: false }” to know its not a kibana routing error? And have the beat check for that, if it exists stop retrying?

**Comment by ph** [comment]
@mattapperson I think that would make sense, because if we return false, its clear that beat cannot recover unless we **enroll** it again and it stop the beat to make unnecessary calls.

**Comment by mattapperson** [comment]
Exactly, and without causing the beat to stop pinging and basically un-enrolling itself if Kibana has an issue
